presto-main/src/main/java/io/prestosql/metadata/FunctionRegistry.java

```diff
@@ -394,13 +393,13 @@ public FunctionRegistry(TypeManager typeManager, BlockEncodingSerde blockEncodin
                 .maximumSize(1000)
                 .expireAfterWrite(1, HOURS)
                 .build(CacheLoader.from(key -> ((SqlScalarFunction) key.getFunction())
-                        .specialize(key.getBoundVariables(), key.getArity(), typeManager, this)));
+                        .specialize(key.getBoundVariables(), key.getArity(), metadata.getTypeManager(), this)));
```

Comment:
Why not keep the field? The code and diff would be smaller.

Reply:
Oops over aggressive refactor